File: color/system_darwin.m
#include "system_darwin.h"

unsigned int convertNSColor(NSColor *color) {
	CGFloat red, green, blue, alpha;
	[[color colorUsingColorSpace:[NSColorSpace deviceRGBColorSpace]] getRed:&red green:&green blue:&blue alpha:&alpha];
	return ((((unsigned int)(255 * alpha + 0.5)) & 0xFF) << 24) |
			((((unsigned int)(255 * red + 0.5)) & 0xFF) << 16) |
			((((unsigned int)(255 * green + 0.5)) & 0xFF) << 8) |
			(((unsigned int)(255 * blue + 0.5)) & 0xFF);
}

unsigned int keyboardFocusColor() {
	return convertNSColor([NSColor keyboardFocusIndicatorColor]);
}

unsigned int selectedTextBackgroundColor() {
	return convertNSColor([NSColor selectedTextBackgroundColor]);
}

unsigned int selectedTextColor() {
	return convertNSColor([NSColor selectedTextColor]);
}

unsigned int textBackgroundColor() {
	return convertNSColor([NSColor textBackgroundColor]);
}

unsigned int textColor() {
	return convertNSColor([NSColor textColor]);
}

File: color/system_darwin.go
package color

import (
	// #cgo CFLAGS: -x objective-c
	// #cgo LDFLAGS: -framework Cocoa
	// #include "system_darwin.h"
	"C"
)

func init() {
	KeyboardFocus = Color(C.keyboardFocusColor())
	SelectedTextBackground = Color(C.selectedTextBackgroundColor())
	SelectedText = Color(C.selectedTextColor())
	TextBackground = Color(C.textBackgroundColor())
	Text = Color(C.textColor())
}
